pkg/log/zap/flags.go

```diff
@@ -60,26 +64,73 @@ func (ev *encoderFlag) Set(flagValue string) error {
 	val := strings.ToLower(flagValue)
 	switch val {
 	case "json":
-		ev.setFunc(newJSONEncoder())
+		ev.setFunc(newJSONEncoder(ecfs))
 	case "console":
-		ev.setFunc(newConsoleEncoder())
+		ev.setFunc(newConsoleEncoder(ecfs))
 	default:
 		return fmt.Errorf("invalid encoder value \"%s\"", flagValue)
 	}
 	ev.value = flagValue
 	return nil
 }
 
-func newJSONEncoder() zapcore.Encoder {
+func newJSONEncoder(ecfs []encoderConfigFunc) zapcore.Encoder {
 	encoderConfig := zap.NewProductionEncoderConfig()
+	for _, f := range ecfs {
+		f(&encoderConfig)
+	}
 	return zapcore.NewJSONEncoder(encoderConfig)
 }
 
-func newConsoleEncoder() zapcore.Encoder {
+func newConsoleEncoder(ecfs []encoderConfigFunc) zapcore.Encoder {
 	encoderConfig := zap.NewDevelopmentEncoderConfig()
+	for _, f := range ecfs {
+		f(&encoderConfig)
+	}
 	return zapcore.NewConsoleEncoder(encoderConfig)
 }
 
+type timeEncoderFlag struct {
+	setFunc func(zapcore.TimeEncoder)
+	value   string
+}
+
+var _ pflag.Value = &timeEncoderFlag{}
+
+func (ev *timeEncoderFlag) String() string {
+	return ev.value
+}
+
+func (ev *timeEncoderFlag) Type() string {
+	return "timeEncoder"
+}
+
+func (ev *timeEncoderFlag) Set(flagValue string) error {
+	ev.setFunc(newTimeEncoder(flagValue))
+	ev.value = flagValue
+	return nil
+}
+
+func newTimeEncoder(flagValue string) zapcore.TimeEncoder {
+	var encoder zapcore.TimeEncoder
+	val := strings.ToLower(flagValue)
+	switch val {
+	case "iso8601", "rfc3339", "rfc3339nano", "millis", "nanos", "epoch":
+		if err := encoder.UnmarshalText([]byte(val)); err != nil {
+			fmt.Printf("not able to unmarshal time encoder value \"%s\"", val)
+		}
+	default:
+		fmt.Printf("invalid time encoder value \"%s\"", val)
```

Comment:
If we pass an invalid string `--zap-time-encoder=foobar` then this prints out an error but returns a nil encoder right? Shouldn't we be returning an error here and above?

Reply:
True, Can we set default to "epoch" time encoder, and print out message saying the same?
```go
	switch val {
	case "iso8601", "rfc3339", "rfc3339nano", "millis", "nanos", "epoch":
		if err := encoder.UnmarshalText([]byte(val)); err != nil {
			fmt.Printf("not able to unmarshal time encoder value \"%s\"", val)
		}
	default:
		encoder = zapcore.TimeEncoder(zapcore.EpochTimeEncoder)
		fmt.Printf("invalid value for time encoder, \"%s\". Setting default to epoch. \n", val)
	}
	return encoder
```

Reply:
I would lean towards returning the error since that's consistent with the behavior for other logging flags.